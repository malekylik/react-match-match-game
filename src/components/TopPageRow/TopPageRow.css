.top__td {
    text-align: center;
    padding: 5px 10px;
    padding: 15px 10px;
}
