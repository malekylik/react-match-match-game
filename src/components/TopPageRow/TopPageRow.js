import React from 'react';
import PropTypes from 'prop-types';

import './TopPageRow.css';

const TopPageRow = ({ number, email, name, score }) => (
    <tr>
        <td className="top__td">
            {number}
        </td>
        <td className="top__td">
            {email}
        </td>
        <td className="top__td">
            {name}
        </td>
        <td className="top__td">
            {score}
        </td>
    </tr>
);

TopPageRow.propTypes = {
    number: PropTypes.number.isRequired, 
    email: PropTypes.string, 
    name: PropTypes.string, 
    score: PropTypes.number, 
};

export default TopPageRow; 
