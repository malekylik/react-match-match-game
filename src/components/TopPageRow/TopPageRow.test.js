import React from 'react';

import { shallow } from 'enzyme';

import TopPageRow from './TopPageRow';

describe('TopPageRow', () => {
    it('Should match snapshot', () => {
        const wrapper = shallow(<TopPageRow number={1} email="test@test.test" name="test" score={10} />);
        expect(wrapper).toMatchSnapshot();
    });
});
