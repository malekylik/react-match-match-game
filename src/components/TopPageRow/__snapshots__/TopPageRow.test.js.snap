// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopPageRow Should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TopPageRow
    email="test@test.test"
    name="test"
    number={1}
    score={10}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <td
          className="top__td"
        >
          1
        </td>,
        <td
          className="top__td"
        >
          test@test.test
        </td>,
        <td
          className="top__td"
        >
          test
        </td>,
        <td
          className="top__td"
        >
          10
        </td>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": 1,
          "className": "top__td",
        },
        "ref": null,
        "rendered": 1,
        "type": "td",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "test@test.test",
          "className": "top__td",
        },
        "ref": null,
        "rendered": "test@test.test",
        "type": "td",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "test",
          "className": "top__td",
        },
        "ref": null,
        "rendered": "test",
        "type": "td",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": 10,
          "className": "top__td",
        },
        "ref": null,
        "rendered": 10,
        "type": "td",
      },
    ],
    "type": "tr",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <td
            className="top__td"
          >
            1
          </td>,
          <td
            className="top__td"
          >
            test@test.test
          </td>,
          <td
            className="top__td"
          >
            test
          </td>,
          <td
            className="top__td"
          >
            10
          </td>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": 1,
            "className": "top__td",
          },
          "ref": null,
          "rendered": 1,
          "type": "td",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "test@test.test",
            "className": "top__td",
          },
          "ref": null,
          "rendered": "test@test.test",
          "type": "td",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "test",
            "className": "top__td",
          },
          "ref": null,
          "rendered": "test",
          "type": "td",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": 10,
            "className": "top__td",
          },
          "ref": null,
          "rendered": 10,
          "type": "td",
        },
      ],
      "type": "tr",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
